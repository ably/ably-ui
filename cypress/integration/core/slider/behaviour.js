describe("Slider", () => {
  const url = "/components/slider?framework=vw";
  const sliderIds = "[data-id='slider']";
  const sliderControlId = "[data-id='slider-controls']";
  const sliderControlNextId = "[data-id='slider-next']";
  const sliderControlPreviousId = "[data-id='slider-previous']";

  const SLIDE_1_COPY =
    "Powers live chat, updates, analytics, and composition for millions of users.";
  const SLIDE_2_COPY =
    "Powers virtual venues for millions of event attendees around the world.";
  const SLIDE_3_COPY =
    "Provides 5 million daily users with live financial commentary and stock tickers.";
  const SLIDE_4_COPY = "Monitors live car performance data across the USA.";

  beforeEach(() => {
    cy.visit(url);
  });

  it("should change the slide back on click", () => {
    cy.viewport("iphone-8");

    cy.get(sliderIds)
      .eq(0)
      .within(() => {
        cy.contains(SLIDE_1_COPY).should("be.visible");
        cy.contains(SLIDE_2_COPY).should("not.be.visible");
        cy.get(sliderControlNextId).click();
        cy.get(sliderControlNextId).click();
        cy.contains(SLIDE_1_COPY).should("not.be.visible");
        cy.contains(SLIDE_2_COPY).should("not.be.visible");
        cy.contains(SLIDE_3_COPY).should("be.visible");
        cy.get(sliderControlPreviousId).click();
        cy.contains(SLIDE_1_COPY).should("not.be.visible");
        cy.contains(SLIDE_2_COPY).should("be.visible");
        cy.contains(SLIDE_3_COPY).should("not.be.visible");
      });
  });

  it("should change the slide forward on click", () => {
    cy.viewport("iphone-8");

    cy.get(sliderIds)
      .eq(0)
      .within(() => {
        cy.contains(SLIDE_1_COPY).should("be.visible");
        cy.contains(SLIDE_2_COPY).should("not.be.visible");
        cy.get(sliderControlNextId).click();
        cy.contains(SLIDE_1_COPY).should("not.be.visible");
        cy.contains(SLIDE_2_COPY).should("be.visible");
      });
  });

  it("should stay on first slide on back click", () => {
    cy.viewport("iphone-8");

    cy.get(sliderIds)
      .eq(0)
      .within(() => {
        cy.contains(SLIDE_1_COPY).should("be.visible");
        cy.contains(SLIDE_2_COPY).should("not.be.visible");
        cy.get(sliderControlPreviousId).click();
        cy.contains(SLIDE_1_COPY).should("be.visible");
        cy.contains(SLIDE_2_COPY).should("not.be.visible");
      });
  });

  it("should stay on last slide on forward click", () => {
    cy.viewport("iphone-8");

    cy.get(sliderIds)
      .eq(0)
      .within(() => {
        cy.contains(SLIDE_1_COPY).should("be.visible");
        cy.contains(SLIDE_2_COPY).should("not.be.visible");
        cy.get(sliderControlNextId).click();
        cy.get(sliderControlNextId).click();
        cy.get(sliderControlNextId).click();
        cy.get(sliderControlNextId).click();
        cy.contains(SLIDE_4_COPY).should("be.visible");
        cy.contains(SLIDE_3_COPY).should("not.be.visible");
        cy.get(sliderControlNextId).click();
        cy.contains(SLIDE_4_COPY).should("be.visible");
        cy.contains(SLIDE_3_COPY).should("not.be.visible");
      });
  });

  it("should enable the slider when the viewport changes", () => {
    cy.viewport("macbook-16");

    cy.get(sliderIds)
      .eq(1)
      .within(() => {
        cy.get(sliderControlId).should("not.be.visible");
      });

    cy.viewport("iphone-8");

    cy.get(sliderIds)
      .eq(1)
      .within(() => {
        cy.get(sliderControlId).should("not.be.visible");
      });
  });
});
