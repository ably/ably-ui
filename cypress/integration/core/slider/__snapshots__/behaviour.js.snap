exports[`Slider > should change the slide forward on click #0`] = `
<ol
  class="grid ui-grid-gap grid-cols-dynamic transform transition-transform "
  data-id="slider-strip"
  style="transform: translateX(184px);"
>
  <li class="" data-id="slider-slide">
    <div class="h-full p-24 bg-white rounded">
      <p class="text-p2 text-center">
        Powers live chat, updates, analytics, and composition for millions of
        users.
      </p>
    </div>
  </li>
  <li class="" data-id="slider-slide">
    <div class="h-full p-24 bg-white rounded">
      <p class="text-p2 text-center">
        Powers virtual venues for millions of event attendees around the world.
      </p>
    </div>
  </li>
  <li class="" data-id="slider-slide">
    <div class="h-full p-24 bg-white rounded">
      <p class="text-p2 text-center">
        Provides 5 million daily users with live financial commentary and stock
        tickers.
      </p>
    </div>
  </li>
  <li class="" data-id="slider-slide">
    <div class="h-full p-24 bg-white rounded">
      <p class="text-p2 text-center">
        Monitors live car performance data across the USA.
      </p>
    </div>
  </li>
</ol>
`;
